SpatialGDK/Source/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -86,7 +86,12 @@ void USpatialWorkerConnection::ConnectToReceptionist(bool bConnectAsClient)
 			{
 				if (OpList->ops[i].op_type == WORKER_OP_TYPE_DISCONNECT)
 				{
-					UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Couldn't connect to SpatialOS: %s"), UTF8_TO_TCHAR(OpList->ops[i].disconnect.reason));
+					FString ErrorMessage(UTF8_TO_TCHAR(OpList->ops[i].disconnect.reason));
+					AsyncTask(ENamedThreads::GameThread, [this, ErrorMessage]
+					{
+						UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Failed to connect to SpatialOS: %s"), *ErrorMessage);
+						this->OnConnectFailed.ExecuteIfBound(ErrorMessage);
```

Comment:
Isn't `OnConnectFailed` accessible without `this->` as you are capturing the this pointer?

Reply:
Maybe you could even try to just capture the delegate by reference in this lambda

Reply:
Agreed, `this->` is not needed. Can't capture `OnConnectFailed` itself though because it's a member variable.